src/components/MapView/utils.ts

```diff
@@ -12,6 +12,28 @@ function getBounds(waypoints: Array<[number, number]>, directionCoordinates: und
     };
 }
 
+function haversineDistance(coordinate1: number[], coordinate2: number[]) {
+    // Radius of the Earth in meters
+    const R = 6371e3;
+    const lat1 = (coordinate1[0] * Math.PI) / 180;
+    const lat2 = (coordinate2[0] * Math.PI) / 180;
+    const deltaLat = ((coordinate2[0] - coordinate1[0]) * Math.PI) / 180;
+    const deltaLon = ((coordinate2[1] - coordinate1[1]) * Math.PI) / 180;
+
+    const a = Math.sin(deltaLat / 2) * Math.sin(deltaLat / 2) + Math.cos(lat1) * Math.cos(lat2) * Math.sin(deltaLon / 2) * Math.sin(deltaLon / 2);
+
+    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
```

Comment:
Could we use a better name for the variable than a or c? If not, maybe it would be helpful to add what it says here or link this https://en.wikipedia.org/wiki/Haversine_formula#The_law_of_haversines

```suggestion
    const a = Math.sin(deltaLat / 2) * Math.sin(deltaLat / 2) + Math.cos(lat1) * Math.cos(lat2) * Math.sin(deltaLon / 2) * Math.sin(deltaLon / 2);
    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
```